pkg/provider/passbolt/passbolt.go

```diff
@@ -0,0 +1,298 @@
+	Logout(ctx context.Context) error
+	GetResource(ctx context.Context, resourceID string) (*api.Resource, error)
+	GetResources(ctx context.Context, opts *api.GetResourcesOptions) ([]api.Resource, error)
+	GetResourceType(ctx context.Context, typeID string) (*api.ResourceType, error)
+	DecryptMessage(message string) (string, error)
+	GetSecret(ctx context.Context, resourceID string) (*api.Secret, error)
+}
+
+func (provider *ProviderPassbolt) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube kclient.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	config := store.GetSpec().Provider.Passbolt
+
+	password, err := resolvers.SecretKeyRef(
+		ctx,
+		kube,
+		store.GetKind(),
+		namespace,
+		config.Auth.PasswordSecretRef,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	privateKey, err := resolvers.SecretKeyRef(
+		ctx,
+		kube,
+		store.GetKind(),
+		namespace,
+		config.Auth.PrivateKeySecretRef,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := api.NewClient(nil, "", config.Host, privateKey, password)
+	if err != nil {
+		return nil, err
+	}
+
+	provider.client = client
+	return provider, nil
+}
+
+func (provider *ProviderPassbolt) SecretExists(_ context.Context, _ esv1beta1.PushSecretRemoteRef) (bool, error) {
+	return true, nil
+}
+
+func (provider *ProviderPassbolt) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	if err := assureLoggedIn(ctx, provider.client); err != nil {
+		return nil, err
+	}
+
+	secret, err := getPassboltSecret(ctx, provider.client, ref.Key)
+	if err != nil {
+		return nil, err
+	}
+
+	if ref.Property == "" {
+		return json.Marshal(secret)
+	}
+
+	return secret.GetProp(ref.Property)
+}
+
+func (provider *ProviderPassbolt) PushSecret(_ context.Context, _ *corev1.Secret, _ esv1beta1.PushSecretData) error {
+	return nil
+}
+
+func (provider *ProviderPassbolt) DeleteSecret(_ context.Context, _ esv1beta1.PushSecretRemoteRef) error {
+	return nil
+}
+
+func (provider *ProviderPassbolt) Validate() (esv1beta1.ValidationResult, error) {
+	return esv1beta1.ValidationResultUnknown, nil
+}
+
+func (provider *ProviderPassbolt) GetSecretMap(_ context.Context, _ esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	return nil, nil
+}
+
+func (provider *ProviderPassbolt) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	res := make(map[string][]byte)
+
+	if ref.Name == nil {
+		return res, nil
+	}
+
+	if ref.Name.RegExp == "" {
+		return res, nil
+	}
+
+	if err := assureLoggedIn(ctx, provider.client); err != nil {
+		return nil, err
+	}
+
+	resources, err := provider.client.GetResources(ctx, &api.GetResourcesOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	nameRegexp, err := regexp.Compile(ref.Name.RegExp)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, resource := range resources {
+		if !nameRegexp.MatchString(resource.Name) {
+			continue
+		}
```

Comment:
This makes the RegExp parameter mandatory since it won't return anything unless you define something like "*.".

But in your API definition the Regex is an optional field.

Reply:
I added an error for the case of a missing `RegExp` parameter.

Please let me know if this is insufficient or if you prefer another approach.

Reply:
Fine by me.